merkletree/proof.go

```diff
@@ -16,6 +23,75 @@ type ProofNode interface {
 	Commitment() []byte
 }
 
+func computeLeafHash(ap *AuthenticationPath,
+	leafIndex, leafCommitment []byte, leafLevel int, isLeafEmpty bool) (leafHash []byte) {
+	if isLeafEmpty {
```

Comment:
> ``ap.Leaf`` is just an interface instance. 

Oh that's right. You're right, we can't use it.

> we have to instantiate new ``AuthenticationPath`` from input arrays 

Isn't this the same issue we were discussing in d46c806 for STRs and TBs? Or is your point that the client doesn't have to re-serialize the AP (like for a signature in the case of the STR)? I know I'm being nitpicky, but I would like to be consistent across the codebase.